### PR TITLE
Reworks Maintenance God into Greytide God (also add a couple of signals + components + optional custom religion descriptions)

### PR DESCRIPTION
**Pull request by Metekillot**
## About The Pull Request

Maintenance God is like the burdened god, except that you only just kill yourself, because there aren't enough mice on station to make enough slurry to get anything besides maintenance adaption and a couple of moldied foods. You also don't get any resistance to the toxin/organ damage from eating anything except an extremely specific subset of gross food.

There are a few other reasons it sucks, so I've reworked it into Greytide God instead. Maintenance adaption stays -- and in fact, I made it severe and permanent to boot -- but in exchange you get toxin lover to facilitate the fact that you're meant to be a constantly drug addled, trash eating maintenance creature. You can also gain favor from turning in a set of Greytide themed sacrifice items.

Toxin lover makes you invert toxin damage -- heal from damage, hurt from healing -- so this also encourages you to use a few of the drugs that cause toxin damage to ludicrous excess, though it won't do anything about profound organ failure besides saving you specifically from liver failure killing you with tox (it will not affect liver failure knocking you out until you get it replaced).

I also added goofy whispers and sounds to Greytide God.

Somewhat related: I added signals to addiction code to facilitate preventing the end -- or the beginning -- of withdrawal with accompanying signals and components, and also a bit of logic to religions to let people give their sacrifice description custom text instead of the autogenerated muck it defaults to.
## Why It's Good For The Game

Maintenance God seems like it's a joke religion except the joke is on you for picking it. I changed it to Greytide God and played the joke up, and gave it TOXIN_LOVER so you can play out the maints dwelling trash man to your heart's content.

### ASK ME HOW GREYTIDE GOD CHANGED MY LIFE
BorisTrailer
## Changelog
:cl: Bisar
add: Maintenance God has been reworked into Greytide God.
add: Alongside organic slurry, Greytide God also desires several items known to be carried by their faithful at all times.
add: Maintenance adaption immediately sets you at the maximum withdrawal for maintenance drugs and locks you there.
add: Maintenance adaption gives you TOXIN_LOVER trait, which reverses the effects of toxin damage. Organ failure will still ruin your day.
code: Added signals for stopping the beginning or end of withdrawal, and signals for the beginning and end of withdrawal.
code: Added optional customized descriptions for religion sacrifice lists, if you don't want the auto-generated one.
/:cl:

### CODE REVIEW

**Comment by carlarctg** [comment]
how is this at all related to greytides

what's the point of the rename if you just enhanced its maintenance lurking theme

greytiding is not being a maint goblin

**Comment by carlarctg** [comment]
add a boon that lets you summon tider stuff like pumpip injectors which are very hars to procure

**Comment by Metekillot** [comment]
> how is this at all related to greytides
> 
> what's the point of the rename if you just enhanced its maintenance lurking theme
> 
> greytiding is not being a maint goblin

giphy

**Comment by Metekillot** [comment]
> add a boon that lets you summon tider stuff like pumpip injectors which are very hars to procure

Sure, summoning tide tools for your faithful or whatever, because you cannot go out into the harsh light of the hallways, swarming with Security Officer casting their hateful red W upon your persecuted person

But as a separate PR if this rework goes throuh

**Comment by Loafin34** [comment]
could there be a 10 cost invoke that lets you summon more mice to the station? so that only one mouse is enough? also i think you can replicate mice with cheese

**Comment by OuroborusReporter** [comment]
The WGW reference as a whisper is horrifying, but other than that and concerns over printing faith by stealing a lathe board, this almost makes being a maint rat chaplain worthwhile.

**Comment by GPeckman** [comment]
The maintenance sect was the least chosen sect per superset, and for good reason. All those maintenance adaptions force you to hide in maints instead of interacting with the rest of the station, which sucks for an RP friendly job like chaplain. Hopefully these changes will make it more popular.

**Comment by Xkallubar** [comment]
No reason to rename the sect, and "Greytide God" is less RP friendly. Being blinded by light is TERRIBLE for tiding.

**Comment by Jacquerel** [comment]
I hate the idea of "greytide god" as a term existing in the game. That's an OOC description of player behaviour, not an in-universe concept.

**Comment by Sadboysuss** [comment]
What if your liver is non-functional while you're a toxin lover? Is that just infinite passive healing? Pretty sure slimepeople don't have this problem cause they don't have a liver (I think?)

File: code/modules/religion/religion_sects.dm [hunk 1]
```diff
@@ -425,14 +427,60 @@
...
+		/obj/item/wirecutters = 15, // doorhacking is patient and kind; it does not rcd or shock.
+		)
+	custom_sacrifice_description = "\
+	organic slurry, and the tools of the tide: yellow gloves... \
+	 Maintenance pump up, but also the baton of those who make it necessary, or your own improvised stunprods. \
+	 Spears: pointed with glass from windows that blocked the way of your gospel. \
+	 Crowbars -- or even the hallowed jaws of life -- and the wirecutters that make them the locus of your power... \
+	 GREYTIDE STATIONWIDE"
+
+	var/list/tide_whispers = list(
```
Comment (Jacquerel) [inline code/modules/religion/religion_sects.dm:456]: I also think this is mostly just pretty cringe tbh

**Comment by Inari-Whitebear** [comment]
> I hate the idea of "greytide god" as a term existing in the game. That's an OOC description of player behaviour, not an in-universe concept.

The term as a whole or greytide itself? Cause greytide exists ingame.. and adding "god" is just saying "they're good at it" anyway

**Comment by MercuryArrow** [comment]
As someone who's been very vocal about the problems with maintenance sect following the last PR that affected it(changing mice dipping behaviour) and has been following plans for it closely, here's my two cents. Any improvement is a welcome improvement as its current state is nigh-unusable, however...

1) This changes the actual flavour and vibe of the sect in a small but meaningful way that clashes a bit with some otherwise fitting stuff like the rogue blade, certain outfits etc. Far from a dealbreaker but since other people have voiced concern about the flavour change, may as well echo that too on my part because thematics are a big part of what make sects interesting and attractive to chaplains.
2) Toxin lover is actually a great idea but as you yourself state in the PR, it doesn't do a ton to offset some of the other issues of both the addiction/withdrawl itself and the likely liver failure, which is almost certainly going to be a problem for maint chaplains given the relatively small quantities the more useful maint drugs tend to come in when made from scratch.
3) The changes to favour accumulation feel a little bit like an overcorrection as much as I hate to say it - it went from you needing like 30 rats per person going through the rites to one stun baton and a couple of common tools allowing the chaplain to rush through the two main rites in potentially minutes if they're willing to make a nuisance of themselves. This feels both far too quick and easy overall and has the unfortunate effect of encouraging some rather griefy behaviours from people wanting to speedrun their way to adaptation+eyes+totem for the full maint goblin experience.
4) On a more meta/codebase-wide level, there was a PR not too long ago that separated out infuser behaviour into something that could be slapped onto other structures with the specific stated intent of being leveraged for a larger-scale maint sect rework down the line. That doesn't mean this PR doesn't have value or we couldn't benefit from it going through, again, any improvements to maint sect are welcome given the state of it right now. It does mean the lifespan of these changes is up in the air though if that larger rework gets pulled sooner or later.

**Comment by Metekillot** [comment]
Eh yeah it needs more work shopping

**Pull request closed without merging**